File: Java/CajaSupermercado/src/GeneradorTicket.java
import java.io.FileWriter;
import java.io.PrintWriter;
import java.io.IOException;

public class GeneradorTicket {

    public void crearArchivoTXT(Carrito CarritoVenta){

    try{
        FileWriter fw = new FileWriter("ticket.txt");
        PrintWriter pw = new PrintWriter(fw);

        pw.println("====================");
        pw.println("  Supermercado poo     ");
        pw.println("====================");

        for (Producto p : CarritoVenta.getLista()){
            pw.println("-" + p.getNombre() + ": $" +p.getPrecio());
            

        } 
        pw.println("--------------------");
            pw.println("   Total a Pagar: $" + CarritoVenta.getTotal());
            pw.println("====================");
            pw.close();
            System.out.println("Archivo generado");


    }catch (IOException e){
        System.out.println("Error I/O crítico" + e.getMessage());
    }
    }
}    

File: Cpp/Tarea1/GeneradorTicket.cpp
#include <fstream>
#include <iostream>
#include <vector>
#include <string>
#include "Carrito.cpp" // Asegúrate de que Carrito incluya a Producto.h

class GeneradorTicket {
public:
    void crearArchivoTXT(const Carrito &carritoVenta) {
        std::ofstream archivo("ticket.txt");
        
        if (!archivo.is_open()) {
            std::cerr << "Error I/O crítico: No se pudo abrir el archivo" << std::endl;
            return;
        }
        
        archivo << "====================" << std::endl;
        archivo << "  Supermercado POO  " << std::endl;
        archivo << "====================" << std::endl;
        
        // Usamos los getters definidos en la clase Producto
        for (const auto& p : carritoVenta.getLista()) {
            archivo << "- " << p->getNombre() << ": $" << p->getPrecio() << std::endl;
        }
        
        archivo << "--------------------" << std::endl;
        archivo << "   Total a Pagar: $" << carritoVenta.getTotal() << std::endl;
        archivo << "====================" << std::endl;
        
        archivo.close();
        std::cout << "Archivo generado con éxito." << std::endl;
    }
};